cmd/opentelemetry/app/exporter/elasticsearchexporter/spanstore.go

```diff
@@ -216,11 +223,36 @@ func hashCode(serviceName, operationName string) string {
 
 type bulkItem struct {
 	// span associated with the bulk operation
-	span *dbmodel.Span
+	spanData esmodeltranslator.ConvertedData
 	// isService indicates that this bulk operation is for service index
 	isService bool
 }
 
 func (w *esSpanWriter) esClientVersion() int {
 	return w.client.MajorVersion()
 }
+
+func bulkItemsToTraces(bulkItems []bulkItem) pdata.Traces {
+	traces := pdata.NewTraces()
+	for _, op := range bulkItems {
+		spanData := op.spanData
+		if traces.ResourceSpans().Len() == 0 {
+			traces.ResourceSpans().Resize(1)
+		} else {
+			traces.ResourceSpans().Resize(traces.ResourceSpans().Len())
```

Comment:
Does this do anything?  

Should we just set `traces.ResourceSpans().Resize(len(bulkItems))` before the loop and make each bulk item its own ResourceSpans?  This would be the simplest way to do it.

Otherwise I think you'd need a map of `Resource` => `ResourceSpans` which you could check for every `bulkItem` to see if a `ResourceSpan` for this `Resource` already existed and append to it.

Reply:
+1 it's better to allocate at the beginning.